main.go

```diff
@@ -320,26 +400,26 @@ func runTests(perfStatsForTest *perfTestUtils.PerfStats, mode int, testSuite *te
 
 	// 1. Start go routine to grab memory in use.
 	// Peak memory is stored in peakMemoryAllocation variable.
-	quit := make(chan bool)
+	chanQuitPkMem := make(chan bool)
 	go func() {
 		for {
 			select {
-			case <-quit:
+			case <-chanQuitPkMem:
 				return
 			default:
 				memoryStatsUrl := "http://" + configurationSettings.TargetHost + ":" + configurationSettings.TargetPort + configurationSettings.MemoryEndpoint
 				resp, err := http.Get(memoryStatsUrl)
 				if err != nil {
 					log.Error("Memory analysis unavailable. Failed to retrieve memory Statistics from endpoint ", memoryStatsUrl, ". Error: ", err)
-					quit <- true
+					chanQuitPkMem <- true
 				} else {
 					body, _ := ioutil.ReadAll(resp.Body)
-					defer resp.Body.Close()
+					resp.Body.Close()
 					m := new(perfTestUtils.Entry)
```

Comment:
Using "defer" in a loop can cause a resource leak. In this case, the goroutine runs for the duration causing the defer action to never invoke, and a subsequent resource leak. XAPI-1327